system/default_ad_api/src/planning.cpp

```diff
@@ -139,19 +141,19 @@ void PlanningNode::on_timer()
         factor.distance = motion_utils::calcSignedArcLength(points, curr_point, stop_point);
       }
     }
-  }
 
-  // Set the status if it is unknown.
-  const auto is_vehicle_stopped = stop_checker_->isVehicleStopped(stop_duration_);
-  for (auto & factor : velocity.factors) {
-    if ((factor.status == VelocityFactor::UNKNOWN) && (!std::isnan(factor.distance))) {
-      const auto is_stopped = is_vehicle_stopped && (factor.distance < stop_distance_);
-      factor.status = is_stopped ? VelocityFactor::STOPPED : VelocityFactor::APPROACHING;
+    // Set the status if it is unknown.
+    const auto is_vehicle_stopped = stop_checker_->isVehicleStopped(stop_duration_);
+    for (auto & factor : velocity.factors) {
+      if ((factor.status == VelocityFactor::UNKNOWN) && (!std::isnan(factor.distance))) {
+        const auto is_stopped = is_vehicle_stopped && (factor.distance < stop_distance_);
+        factor.status = is_stopped ? VelocityFactor::STOPPED : VelocityFactor::APPROACHING;
+      }
     }
-  }
 
-  pub_velocity_factors_->publish(velocity);
-  pub_steering_factors_->publish(steering);
+    pub_velocity_factors_->publish(velocity);
+    pub_steering_factors_->publish(steering);
+  }
```

Comment:
What do you intend to execute them if `(trajectory_ && kinematic_state_) ` which originally executed regardless of `trajectory_` and `kinematic_state_`.

Reply:
Apologies for any confusion.  I misunderstood that published topics in `on_timer` are related to subscribed topics directly. I will revert to the original implementation.